mso/resource_mso_schema_template_contract_service_graph.go

```diff
@@ -90,608 +72,176 @@ func resourceTemplateContractServiceGraph() *schema.Resource {
+	if err != nil {
+		return err
 	}
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+	return resourceMSOSchemaTemplateContractServiceGraphRead(d, m)
+}
 
-	if foundSite && foundTemp {
-		length := len(temprelationList)
-		nodeList := make([]interface{}, 0, 1)
-		for i := 0; i < length; i++ {
-			tempMap := temprelationList[i].(map[string]interface{})
-			siteMap := siterelationList[i].(map[string]interface{})
-
-			allMap := make(map[string]interface{})
-			allMap["provider_connector_bd_name"] = tempMap["provider_connector_bd_name"]
-			allMap["consumer_connector_bd_name"] = tempMap["consumer_connector_bd_name"]
-
-			tp := strings.Split(siteMap["provider_connector_cluster_interface"].(string), "/")
-			token := strings.Split(tp[len(tp)-1], "-")
-			allMap["provider_connector_cluster_interface"] = token[1]
-
-			tp = strings.Split(siteMap["consumer_connector_cluster_interface"].(string), "/")
-			token = strings.Split(tp[len(tp)-1], "-")
-			allMap["consumer_connector_cluster_interface"] = token[1]
-
-			if siteMap["provider_connector_redirect_policy"] != nil {
-				tp := strings.Split(siteMap["provider_connector_redirect_policy"].(string), "/")
-				token1 := strings.Split(tp[1], "-")
-				allMap["provider_connector_redirect_policy_tenant"] = token1[1]
-
-				token2 := strings.Split(tp[len(tp)-1], "-")
-				allMap["provider_connector_redirect_policy"] = token2[1]
-			}
-			if siteMap["consumer_connector_redirect_policy"] != nil {
-				tp := strings.Split(siteMap["consumer_connector_redirect_policy"].(string), "/")
-				token1 := strings.Split(tp[1], "-")
-				allMap["consumer_connector_redirect_policy_tenant"] = token1[1]
-
-				token2 := strings.Split(tp[len(tp)-1], "-")
-				allMap["consumer_connector_redirect_policy"] = token2[1]
-			}
-
-			if siteMap["provider_subnet_ips"] != nil {
-				allMap["provider_subnet_ips"] = siteMap["provider_subnet_ips"]
-			}
-			if siteMap["consumer_subnet_ips"] != nil {
-				allMap["consumer_subnet_ips"] = siteMap["consumer_subnet_ips"]
-			}
-
-			nodeList = append(nodeList, allMap)
-		}
-		d.Set("schema_id", schemaId)
-		d.Set("site_id", siteID)
-		d.Set("template_name", templateName)
-		d.Set("node_relationship", nodeList)
-		d.Set("contract_name", contractName)
-
-		if d.Get("service_graph_name") == serviceGraph {
-			d.SetId(serviceGraph)
-		} else {
-			d.SetId("")
-			return nil, fmt.Errorf("No service graph found for given name")
-		}
-	} else {
-		d.SetId("")
-		return nil, fmt.Errorf("No service graph found for given name")
+func resourceMSOSchemaTemplateContractServiceGraphRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] Begining Read Template Contract Service Graph")
+	serviceGraphTokens := strings.Split(d.Id(), "/")
+	d.Set("schema_id", serviceGraphTokens[1])
+	d.Set("template_name", serviceGraphTokens[3])
+	d.Set("contract_name", serviceGraphTokens[5])
+	err := setSchemaTemplateContractServiceGraphAttrs(d, m, false)
+	if err != nil {
+		return err
```

Comment:
do we need the errorForObjectNotFound function here?

Reply:
errorForObjectNotFound used in the setSchemaTemplateContractServiceGraphAttrs function.

Reply:
I did a search and in all other resources and data sources this is only called from the read function. Should we keep this consistent to ensure we are not introducing any undesired behaviour?

Reply:
Sure I will move back to the read function.